puzzler.py

```diff
@@ -215,6 +215,15 @@ def generate_puzzles(instream, outstream, engine, variant, depth, win_threshold,
             continue
 
         if len(pv) > stm_index and (not mate_only or (len(types) > 0 and types[0] == 'mate')):
+            # Check for 'tsume' condition
+            is_tsume_puzzle = True
+            for i in range(stm_index, len(pv), 2): # Iterate through player's moves
+                moves_to_check = pv[:i+1]
+                san_moves = sf.get_san_moves(current_variant, fen, moves_to_check)
+                if not san_moves or '+' not in san_moves[-1]:
```

Comment:
Thanks for the PR.

There is `gives_check` in pyffish, which I think would be cleaner here as it exposes the same logic, just more directly.